src/components/TodoModal/TodoModal.tsx

```
@@ -1,42 +1,72 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { setCurrentTodo } from '../../features/currentTodo';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    if (!currentTodo) {
+      return;
+    }
+
+    getUser(currentTodo.userId)
+      .then(setUser)
+      .finally(() => setLoading(false));
```

There is no error handling for the `getUser(...)` call. If the request fails the error will be swallowed and only `.finally` will run, which makes debugging harder. Add a `.catch(...)` to log or handle the error before `.finally(...)`. Example:

```
getUser(currentTodo.userId)
  .then(setUser)
  .catch(err => console.error('Failed to load user for todo', currentTodo.id, err))
  .finally(() => setLoading(false));
```

Add this catch between the `.then` (line 20) and `.finally` (line 21) so failures are visible in the console and tests.